compliance_checker/cf/util.py

```diff
@@ -327,15 +328,28 @@ def download_cf_standard_name_table(version, location=None):
     if location is None:  # This case occurs when updating the packaged version from command line
         location = resource_filename('compliance_checker', 'data/cf-standard-name-table.xml')
 
-    url = "http://cfconventions.org/Data/cf-standard-names/{0}/src/cf-standard-name-table.xml".format(version)
-    r = requests.get(url, allow_redirects=True)
-    if r.status_code == 200:
-        print("Downloading cf-standard-names table version {0} from: {1}".format(version, url), file=sys.stderr)
-        with open(location, 'wb') as f:
-            f.write(r.content)
+    if version == 'latest':
+        from bs4 import BeautifulSoup
+        from urllib.parse import urljoin
+        tables_req = requests.get("http://cfconventions.org/documents.html")
+        tables_req.raise_for_status()
+        # use bs4 due to HTML parse errors under lxml
+        html_content = BeautifulSoup(tables_req.text, features='lxml')
+        # the first a element found with href ending in
+        # "cf-standard-name-table.xml should be the latest CF table
+        html_elem = html_content.find('a',
+                     {'href': lambda s: s.endswith("cf-standard-name-table.xml")})
+        url = urljoin("http://cfconventions.org", html_elem.attrs['href'])
     else:
```

Comment:
What is the behavior when users request `... -d banana-town`? Does this have an ugly error on invalid table versions?

Reply:
Requests would try to fetch the version from the CF Conventions webpage and it would 404.  `raise_for_status()` then raises an exception.